x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -500,62 +555,84 @@ export class ManifestManager {
-
-              try {
-                await this.packagePolicyService.update(
-                  this.savedObjectsClient,
-                  // @ts-expect-error TS2345
-                  undefined,
-                  id,
-                  newPackagePolicy
-                );
-                this.logger.debug(
-                  `Updated package policy ${id} with manifest version ${manifest.getSemanticVersion()}`
-                );
-              } catch (err) {
-                errors.push(err);
-              }
-            } else {
-              this.logger.debug(
-                `No change in manifest content for package policy: ${id}. Staying on old version`
-              );
-            }
+      (packagePoliciesBatch) => {
+        allPackagePolicies.push(...packagePoliciesBatch);
+      }
+    );
+
+    const packagePoliciesToUpdate: PackagePolicy[] = [];
+
+    const errors: Error[] = [];
+    allPackagePolicies.forEach((packagePolicy) => {
+      const { id } = packagePolicy;
+      if (packagePolicy.inputs.length > 0 && packagePolicy.inputs[0].config !== undefined) {
+        const oldManifest = packagePolicy.inputs[0].config.artifact_manifest ?? {
+          value: {},
+        };
+
+        const newManifestVersion = manifest.getSemanticVersion();
+        if (semver.gt(newManifestVersion, oldManifest.value.manifest_version)) {
+          const serializedManifest = manifest.toPackagePolicyManifest(id);
+
+          if (!manifestDispatchSchema.is(serializedManifest)) {
+            errors.push(new EndpointError(`Invalid manifest for policy ${id}`, serializedManifest));
+          } else if (!manifestsEqual(serializedManifest, oldManifest.value)) {
+            packagePolicy.inputs[0].config.artifact_manifest = { value: serializedManifest };
+            packagePoliciesToUpdate.push(packagePolicy);
           } else {
-            this.logger.debug(`No change in manifest version for package policy: ${id}`);
+            this.logger.debug(
+              `No change in manifest content for package policy: ${id}. Staying on old version`
+            );
           }
         } else {
-          errors.push(
-            new EndpointError(`Package Policy ${id} has no 'inputs[0].config'`, newPackagePolicy)
-          );
+          this.logger.debug(`No change in manifest version for package policy: ${id}`);
         }
+      } else {
+        errors.push(
+          new EndpointError(`Package Policy ${id} has no 'inputs[0].config'`, packagePolicy)
+        );
       }
+    });
+
+    // Split updates in batches with batch size: packagerTaskPackagePolicyUpdateBatchSize
+    const updateBatches = chunk(
+      packagePoliciesToUpdate,
+      this.packagerTaskPackagePolicyUpdateBatchSize
     );
 
+    for (const currentBatch of updateBatches) {
+      const response = await this.packagePolicyService.bulkUpdate(
+        this.savedObjectsClient,
+        // @ts-expect-error TS2345
```

Comment:
Any way we can get around this?  Do we have an (internal) ESclient we can provide to the service method?

not doing so, when it is a required dependency of this service method, will likely leave us exposed to bugs in the future and possibly make fleet unstable. I checked the code in fleet for this method and it passes the `esClient` on to other services, the most important one (i think) is the one that bumps the revision number on the Agent policy.

Reply:
Will revisit this and see if we can use any esClient from our service here. Note that this was done in this way before with the `.update`, but good point on ensuring this works to avoid future bugs

Reply:
https://github.com/elastic/kibana/pull/160387/commits/92b521b0b0edb8a0ded51f05f0880d42d78cc259